### PR TITLE
nextpvr: add default settings file

### PR DESCRIPTION
**Pull request by emveepee**
Kodi doesn't pass environment variables to the startup script unless the user saves a configuration via settings configuration, even if it is defaults.  The service starts before the user has a chance to save settings and the setting for the NextPVR service are optional.

<del>Ideally Kodi should pass the default values from the global settings.xml</del>  

As pointed out by @mglae the reason for this issue is the addon wasn't including settings-default.xml.  This PR adds the required file.

### CODE REVIEW

**Comment by chewitt** [comment]
@emveepee please change the commit message to follow house-style, e.g. `nextpvr: fix for variables not initialized by Kodi` or `nextpvr: adjust service startup variables` .. thanks

**Comment by emveepee** [comment]
I will need to change this since the LE build system ignores the version number using this package.mk, so I am holding off on the backport.  I will need to separate the PKG_ADDON_VERSION from the version being downloaded.

**Comment by mglae** [comment]
> Kodi doesn't pass environment variables to the startup script unless the user saves a configuration via settings configuration, even if it is defaults.

There is support for [settings-default.xml](https://github.com/LibreELEC/LibreELEC.tv/blob/4a8b551af442a8385e8b8e0344a76f02fb669d0f/packages/mediacenter/kodi/profile.d/00-addons.conf#L14) in oe_setup_addon to always have the defaults available.

**Comment by emveepee** [comment]
> > Kodi doesn't pass environment variables to the startup script unless the user saves a configuration via settings configuration, even if it is defaults.
> 
> There is support for [settings-default.xml](https://github.com/LibreELEC/LibreELEC.tv/blob/4a8b551af442a8385e8b8e0344a76f02fb669d0f/packages/mediacenter/kodi/profile.d/00-addons.conf#L14) in oe_setup_addon to always have the defaults available.

Thanks that works like a charm.

File: packages/addons/service/nextpvr/source/bin/nextpvr-downloader
Comment (CvH) [inline packages/addons/service/nextpvr/source/bin/nextpvr-downloader:12]: is this construct just to transform `6.1.1~1` to `6.1.1` ?

untested
```
NEXTPVR_VERSION="@NEXTPVR_VERSION@"
NEXTPVR_FILE="NPVR-${NEXTPVR_VERSION%~*}.zip"
```

Comment (emveepee) [reply packages/addons/service/nextpvr/source/bin/nextpvr-downloader]: Yes, but your solution is better, I didn't know that form.  Updated.

**Comment by CvH** [comment]
> @emveepee please change the commit message to follow house-style, e.g. `nextpvr: fix for variables not initialized by Kodi` or `nextpvr: adjust service startup variables` .. thanks

@emveepee ^^^^^^

**Comment by emveepee** [comment]
> > @emveepee please change the commit message to follow house-style, e.g. `nextpvr: fix for variables not initialized by Kodi` or `nextpvr: adjust service startup variables` .. thanks
> 
I did change the commit message, I had changed the title of the PR earlier.  I disagree on general principle to the style of including the addon name in each commit.

**Comment by chewitt** [comment]
@emveepee We insist on house-style for commit messages because we human-generate the change lists used in release notes and blog posts; and when scanning hundreds of commits in a branch it helps us to spot the important changes and ignore non-core things like add-on updates. In a single-app repo it's not needed because all changes are about that one app. In a repo with 380+ app packages (plus plumbing for the build-system) it's important to know what was touched.

**Review by chewitt — APPROVED** [review]

**Comment by emveepee** [comment]
> @emveepee We insist on house-style for commit messages because we human-generate the change lists used in release notes and blog posts; and when scanning hundreds of commits in a branch it helps us to spot the important changes and ignore non-core things like add-on updates. In a single-app repo it's not needed because all changes are about that one app. In a repo with 380+ app packages (plus plumbing for the build-system) it's important to know what was touched.

Sorry I was thinking about a PR with multiple commits I thought you only would have used the PR title and message.

**Comment by chewitt** [comment]
The general goal is "one logical change per commit" so if a PR touches multiple things having multiple commits is not an issue at all. Sometimes it also makes sense to have multiple commits for easier review of changes, but then we'll squash everything to a single commit before merging to keep the changelog tidy. And othertimes we forget, or ignore the rule, or whatever .. but we're quite firm on the house-style rule as it does help with finding stuff :)

**Comment by emveepee** [comment]
Seeing the github slack I understand better.  I would prefer multiple commits in on PR, since the service package has  multiple unrelated components, build, install/download, service startup, and settings GUI but I will get used to it.  Thanks.

**Pull request merged**
